src/vm/eventpipeprotocolhelper.cpp

```diff
@@ -78,41 +154,29 @@ void EventPipeProtocolHelper::StopTracing(IpcStream *pStream)
     }
     CONTRACTL_END;
 
-    uint32_t nNumberOfBytesRead = 0;
-    EventPipeSessionID sessionId = (EventPipeSessionID) nullptr;
-    bool fSuccess = pStream->Read(&sessionId, sizeof(sessionId), nNumberOfBytesRead);
-    if (!fSuccess || nNumberOfBytesRead != sizeof(sessionId))
+    const EventPipeStopTracingCommandPayload* payload = message.TryParsePayload<EventPipeStopTracingCommandPayload>();
+    if (payload == nullptr)
     {
-        // TODO: Add error handling.
+        DiagnosticsIpc::IpcMessage errorMessage(EventPipeErrorHeader, DiagnosticsIpc::ServerErrorPayload{ DiagnosticsIpc::DiagnosticServerErrorCode::BadEncoding });
+        errorMessage.Send(pStream);
         delete pStream;
         return;
     }
 
-    EventPipe::Disable(sessionId);
-    uint32_t nBytesWritten = 0;
-    fSuccess = pStream->Write(&sessionId, sizeof(sessionId), nBytesWritten);
-    if (!fSuccess)
-    {
-        // TODO: Add error handling.
-        delete pStream;
-        return;
-    }
+    EventPipe::Disable(payload->sessionId);
+
+    DiagnosticsIpc::IpcMessage stopTracingResponse(EventPipeSuccessHeader, payload->sessionId);
+    stopTracingResponse.Send(pStream);
```

Comment:
Don't we need to do error handling here when this returns `false`? It seems like none of the callers of this method (`IpcMessage::Send`) is actually using the boolean return value to handle errors.

Reply:
The spec dictates that the connection should be closed after an error message is attempted to be sent.  In those cases, the connection is being closed regardless of the success of the message being sent.  In success cases, however, you're right.  There isn't a "back out path" for failure to write the response.  We may want to consider `Send` failing to be a fatal error and attempt to send an error message and close the connection.  Thoughts?

Reply:
For now, since we are actively closing the connection after every `Send` via `delete pStream`, we will simply treat a failure to send as a client side error.  A well-formed client will end up attempting to read beyond the end of the stream or run into some other "end of stream" exception that will indicate failure.  The server should be resilient to a failed send in both success and error paths.